lib/chromedriver.js

```diff
@@ -625,11 +625,19 @@ class Chromedriver extends events.EventEmitter {
     }
 
     if (this.adb) {
-      log.debug(`Cleaning any old adb forwarded port socket connections`);
+      const udidIndex = this.adb.executable.defaultArgs.findIndex((item) => item === '-s');
+      const udid = udidIndex > -1 ? this.adb.executable.defaultArgs[udidIndex + 1] : null;
+
+      if (udid) {
+        log.debug(`Cleaning this device's adb forwarded port socket connections: ${udid}`);
+      } else {
+        log.debug(`Cleaning any old adb forwarded port socket connections`);
+      }
+
       try {
         for (let conn of await this.adb.getForwardList()) {
           // chromedriver will ask ADB to forward a port like "deviceId tcp:port localabstract:webview_devtools_remote_port"
-          if (conn.indexOf('webview_devtools') !== -1) {
+          if (conn.indexOf('webview_devtools') !== -1 && conn.indexOf(udid) !== -1) {
             let params = conn.split(/\s+/);
             if (params.length > 1) {
               await this.adb.removePortForward(params[1].replace(/[\D]*/, ''));
```

Comment:
I would say it makes sense to perform the cleanup on the given port rather by udid. E.g The logic would be the following:

- check if the `port` that we are going to acquire is open or not
- if the port is closed then simply continue
- if the port is open then list forwarded connections and filter out these that occupy that port
- if no forwards found then raise an error
- if there is a port forward then delete it and check if the port is closed now
- if the port is still open then raise an error otherwise continue

See https://github.com/appium/appium-xcuitest-driver/blob/655964a3d0bb1bc050f0c5aa2805b00110b917eb/lib/device-connections-factory.js#L180

Reply:
> I would say it makes sense to perform the cleanup on the given port rather by udid. E.g. the logic would be the following:
> 
> * check if the `port` that we are going to acquire is open or not
> * if the port is closed then simply continue
> * if the port is open then list forwarded connections and filter out these that occupy that port
> * if no forwards found then raise an error
> * if there is a port forward then delete it and check if the port is closed now
> * if the port is still open then raise an error otherwise continue
> 
> See https://github.com/appium/appium-xcuitest-driver/blob/655964a3d0bb1bc050f0c5aa2805b00110b917eb/lib/device-connections-factory.js#L180

Do you meaning cleanup by `systemPort`？I agree with your opinion. But If I restart the server, last forward process will not be removed as last systemPort is forgot. Second, the old logic only use `webview_devtools` to judge. I don't want to change it so much.